pkg/kubelet/rkt/rkt.go

```diff
@@ -1521,16 +1523,215 @@ func (r *Runtime) RemoveImage(image kubecontainer.ImageSpec) error {
+		}
+
+		cs := kubecontainer.ContainerStatus{
+			ID:    buildContainerID(&containerID{uuid: latestPod.Id, appName: app.Name}),
+			Name:  app.Name,
+			State: appStateToContainerState(app.State),
+			// TODO(yifan): Use the creation/start/finished timestamp when it's implemented.
+			CreatedAt: latestCreationTime,
+			StartedAt: latestCreationTime,
+			ExitCode:  int(app.ExitCode),
+			Image:     app.Image.Name,
+			ImageID:   app.Image.Id,
+			Hash:      hashNum,
+			// TODO(yifan): Note that now all apps share the same restart count, this might
+			// change once apps don't share the same lifecycle.
+			// See https://github.com/appc/spec/pull/547.
+			RestartCount: latestRestartCount,
+			// TODO(yifan): Add reason and message field.
+		}
+
+		podStatus.ContainerStatuses = append(podStatus.ContainerStatuses, &cs)
+
+	}
+
+	return podStatus, nil
 }
 
-func (r *Runtime) ConvertPodStatusToAPIPodStatus(_ *api.Pod, _ *kubecontainer.PodStatus) (*api.PodStatus, error) {
-	return nil, fmt.Errorf("Not implemented yet")
+// TODO(yifan): Delete this function when the logic is moved to kubelet.
+func (r *Runtime) ConvertPodStatusToAPIPodStatus(pod *api.Pod, status *kubecontainer.PodStatus) (*api.PodStatus, error) {
+	// Pod status is not found, this means the pod is not started,
+	// or being restarted while all previous pods are removed (very unlikely).
+	if status == nil {
+		var apiPodStatus api.PodStatus
+		for _, c := range pod.Spec.Containers {
+			cs := api.ContainerStatus{
+				Name:  c.Name,
+				Image: c.Image,
+				// TODO(yifan): Add reason and message.
+				State: api.ContainerState{Waiting: &api.ContainerStateWaiting{}},
+			}
+			apiPodStatus.ContainerStatuses = append(apiPodStatus.ContainerStatuses, cs)
+		}
+		return &apiPodStatus, nil
+	}
+
+	apiPodStatus := &api.PodStatus{
+		// TODO(yifan): Add reason and message field.
+		PodIP: status.IP,
+	}
+
+	for _, c := range status.ContainerStatuses {
+		var st api.ContainerState
+		switch c.State {
+		case kubecontainer.ContainerStateRunning:
+			st.Running = &api.ContainerStateRunning{
+				StartedAt: unversioned.NewTime(c.StartedAt),
+			}
+		case kubecontainer.ContainerStateExited:
+			if pod.Spec.RestartPolicy == api.RestartPolicyAlways ||
+				pod.Spec.RestartPolicy == api.RestartPolicyOnFailure && c.ExitCode != 0 {
+				// TODO(yifan): Add reason and message.
+				st.Waiting = &api.ContainerStateWaiting{}
+			} else {
+				st.Terminated = &api.ContainerStateTerminated{
+					ExitCode:  c.ExitCode,
+					StartedAt: unversioned.NewTime(c.StartedAt),
+					// TODO(yifan): Add reason, message, finishedAt, signal.
+					ContainerID: c.ID.String(),
+				}
+			}
+		default:
+			// Unknown state.
+			// TODO(yifan): Add reason and message.
+			st.Waiting = &api.ContainerStateWaiting{}
+		}
+		cs := api.ContainerStatus{
+			Name:         c.Name,
+			Image:        c.Image,
+			ImageID:      c.ImageID,
+			ContainerID:  c.ID.String(),
+			RestartCount: c.RestartCount,
+			State:        st,
+		}
```

Comment:
@yifan-gu @yujuhong LastTerminationState hasn't been set here. What is LastTermination used for? Just for introspection or there is some functionality relies on it?